apps/backend/src/services/cache_service.py

```
@@ -0,0 +1,503 @@
+            # Base query
+            base_query = select(CacheMetadata).where(CacheMetadata.created_at >= cutoff_time)
+            
+            if data_type:
+                base_query = base_query.where(CacheMetadata.data_type == data_type)
+
+            # Get performance metrics by tier - simplified for current schema
+            performance_by_tier = [
+                {
+                    "tier": "default",
+                    "avg_hit_rate": 0,  # Not tracked in current schema
+                    "avg_access_count": total_accesses / max(1, total_entries),
```

Ruff (0.12.2)

Undefined name `total_accesses`

(F821)